query_optimizer/ExecutionGenerator.cpp

```diff
@@ -511,15 +573,35 @@ void ExecutionGenerator::convertSelection(
   const QueryContext::insert_destination_id insert_destination_index =
       query_context_proto_->insert_destinations_size();
   S::InsertDestination *insert_destination_proto = query_context_proto_->add_insert_destinations();
-  createTemporaryCatalogRelation(physical_selection,
-                                 &output_relation,
-                                 insert_destination_proto);
 
   // Create and add a Select operator.
   const CatalogRelationInfo *input_relation_info =
       findRelationInfoOutputByPhysical(physical_selection->input());
   DCHECK(input_relation_info != nullptr);
 
+  bool is_temp_relation_partitioned = false;
+  P::TableReferencePtr table_reference;
+  if (P::SomeTableReference::MatchesWithConditionalCast(physical_selection->input(), &table_reference) &&
+      table_reference->relation()->hasPartitionScheme()) {
+    const attribute_id partition_attr_id =
+       table_reference->relation()->getPartitionScheme().getPartitionSchemeHeader().getPartitionAttributeId();
+    const E::AttributeReferencePtr &attr = table_reference->attribute_list()[partition_attr_id];
+    if (E::ContainsExpression(physical_selection->getOutputAttributes(), attr)) {
```

Comment:
I was wondering what happens if the input relation uses the range-based partition. I think we could suffer from a repartition from a range-based to a hash-based, right?